### PR TITLE
docs(buttons): Adds React documentation for Button component

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

**Comment by diegonvs** [comment]
Depends on #2063

**Comment by diegonvs** [comment]
Depends on #2063

File: next.clayui.com/content/docs/components/buttons.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+packageNpm: "@clayui/button"
+packageVersion: "3.0.0"
+packageStatus: "Beta"
+---
+
+import {ButtonBlock, ButtonDisplayTypes, ButtonGroup, ButtonMonospaced, ButtonWithStar, ButtonSmall} from '../../../src/components/clay/Button';
+
+<div class="nav-toc">
+
+- [Variants and Best Pratices](#variants-and-best-pratices)
```
Comment (matuzalemsteles) [inline next.clayui.com/content/docs/components/buttons.mdx]: I'm not sure if we want to add all the examples and use cases in the documentation, we're half describing again what the API is saying below. Maybe adding just one example of the button and one of the group is enough.

Comment (bryceosterhaus) [reply next.clayui.com/content/docs/components/buttons.mdx]: I agree, I think the API should describe it enough and then we can also have one or two basic examples for demo purposes. I don't think we need examples of every variation and best practice though.

File: next.clayui.com/content/docs/components/buttons.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    - [Button](#button)
+    - [Button Group](#button-group)
+
+</div>
+
+<div class="clay-site-alert alert alert-info">
+	Check the <a href="https://liferay.design/lexicon/">Lexicon</a> <a href="https://liferay.design/lexicon/core-components/buttons/">Button Pattern</a> for a more in-depth look at the motivations and proper usage of this component.
+</div>
+
+## Variants and Best Pratices
```
Comment (bryceosterhaus) [inline next.clayui.com/content/docs/components/buttons.mdx]: A lot of this section seems the same as API

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by matuzalemsteles** [comment]
Resent #2088

**Pull request closed without merging**